src/controllers/mainController.ts

```diff
@@ -252,6 +258,45 @@ export default class MainController implements vscode.Disposable {
         });
     }
 
+    /**
+     * Helper to script a node based on the script operation
+     */
+    public async scriptNode(node: TreeNodeInfo, operation: ScriptOperation, executeScript: boolean = false): Promise<void> {
+        let actionPromise = new Promise<boolean>(async (resolve, reject) => {
+            const nodeUri = ObjectExplorerUtils.getNodeUri(node);
+            let connectionCreds = Object.assign({}, node.connectionCredentials);
+            const databaseName = ObjectExplorerUtils.getDatabaseName(node);
+            // if not connected or different database
+            if (!this.connectionManager.isConnected(nodeUri) ||
+                connectionCreds.database !== databaseName) {
+                // make a new connection
+                connectionCreds.database = databaseName;
+                if (!this.connectionManager.isConnecting(nodeUri)) {
```

Comment:
If it is connecting shouldn't you wait on it to finish connecting before you proceed?